build/xtask/src/config.rs

```diff
@@ -287,6 +288,81 @@ impl Config {
             })
             .collect()
     }
+
+    /// Calculates the output region which contains the given address
+    pub fn output_region(&self, vaddr: u64) -> Option<&str> {
+        let vaddr: u32 = match vaddr.try_into() {
+            Ok(v) => v,
+            Err(_) => return None,
+        };
+        self.outputs
+            .iter()
+            .find(|(_name, out)| {
+                vaddr >= out.address && vaddr < out.address + out.size
+            })
+            .map(|(name, _out)| name.as_str())
```

Comment:
Just a note, `find` returns `Option`, so you don't have to use `map` here if you don't want to (your current use is fine if you prefer it). You could write:

```rust
let name = self.outputs
    .iter()
    .find(stuff)?;
Some(name.as_str())
```